surface/model_openapiv3.go

```diff
@@ -407,12 +407,17 @@ func (b *OpenAPI3Builder) typeForSchema(schema *openapiv3.Schema) (kind FieldKin
 		}
 	}
 	if schema.AdditionalProperties != nil {
-		additionalProperties := schema.AdditionalProperties
-		if propertySchema := additionalProperties.GetSchemaOrReference().GetReference(); propertySchema != nil {
-			if ref := propertySchema.XRef; ref != "" {
-				return FieldKind_MAP, "map[string]" + typeForRef(ref), ""
-			}
+		schemarOrReference := schema.AdditionalProperties.GetSchemaOrReference()
```

Comment:
nit: could you change this to "schemaOrReference" (remove the extra 'r')?

Reply:
Done